graders/python/python_autograder/pl_main.py

```diff
@@ -135,7 +134,7 @@ def add_files(results):
 
         with open(output_fname, mode="w", encoding="utf-8") as out:
             json.dump(grading_result, out)
-    except:
+    except:  # noqa: E722
```

Comment:
Is the bare except needed here? Not sure if it is intended to catch system interrupts.

Reply:
Off the top of my head, I'm actually not sure how often this branch gets taken, if at all.  I'd be inclined to leave it in for now just in case something breaks inside of the grader.

Reply:
Well the issue isn't quite whether this branch gets taken; we would change this to `except Exception`, since otherwise the bare `except` this will catch system interrupts (like a user hitting Ctrl+C). This is why this is a linter issue rather than a style one, see [here](https://www.flake8rules.com/rules/E722.html) for more information.

That was what I was getting at with the original question, it doesn't seem like this code needs to catch signals from the OS to get errors from the grader code.

Reply:
Looking at this again, I think it's _probably_ not going to cause issues to leave in this bare except (since all that will happen is marking a grade from the autograder before exiting), and it's probably best to limit this PR to changes just to make the linter pass.

Taking a look through the autograder code, there's enough little hiccups that I think it would be worth doing a style pass to enable typechecking, and we can revisit this exception then.